### DISCUSSION TITLE
Work required to extend from bipedal to quadrupedalism

### DISCUSSION THREAD

**Discussion by Utopiah** [boytchev/mannequin.js#14] [Q&A]
I'm wondering how much work is needed to extend the Mannequin class to have e.g dog or cat.
I naively assume the hierarchy for the bones is the same so maybe only parametrization is needed but maybe that is also practically impossible right now due to some assumptions that are not evident.

I understand that the primary goal for this library was COVID and thus focusing solely on human but I'm wondering if being more general could make the library reach a broader audience.

**Comment by boytchev** [comment]
Hi -

There are different aspects regarding how much work is needed. Right now, I can think of three:
* **skeleton** – the underlying structure is almost the same (as in all vertebrates). Proportions of body parts can be easily changed. However, some structural changes are also required: (1) adding a tail of 4+ bones; (2) making much more flexible spine – currently the spine contains two elements, but for dogs and cats 4+ might be more appropriate); (3) if motions of ears is needed, then they should also be represented in the skeleton. **Difficulty level: 2 of 5** (1=least difficult, 5=most difficult).
* **posture** – the default posture is different in animals, as well as the range of limbs motions. **Difficulty level: 1 of 5**.
* **shape** – currently the shape of body parts is sculptured programmatically, because the underlying idea is not to rely on external 3D models (apart from three.js, everything else is self-contained in a single file). Shaping animal's body part is more like an artistic challenge and personally I consider it the most difficult part. **Difficulty level: 4 of 5**.

The bottom line is that it might not be difficult to adjust proportions and body position of a mannequin to look like a 4-leg animal, but the shape of the body, the limbs and especially the head, might take more time to model.

\- Pavel

PS. Mannequin.js is unrelated to COVID-19. The library evolved from a homework assignment that I give to students enrolled in my "Fundamentals of Computer Graphics" course.
